path_supplier/graphhopper_path_supplier.go

```diff
@@ -0,0 +1,92 @@
+package path_supplier
+
+import (
+	"encoding/json"
+	"fmt"
+	m "github.com/harfmuf/geolocation-generator/model"
+	"io/ioutil"
+	"net/http"
+)
+
+// precision to around .0001 deg, which corresponds to max 10m
+const URL_FORMAT = "%s%s?point=%.4f,%.4f&point=%.4f,%.4f&vehicle=%s&key=%s&points_encoded=false"
+const GRAPHHOPPER_API_URL_BASE = "https://graphhopper.com/api/1"
+const ROUTING_API_URL = "/route"
+
+type GraphHopperPathSuplier struct {
+	baseUrl string
+	apiKey  string
+}
+
+type GraphHopperResponse struct {
+	Paths []Path
+}
+
+type Path struct {
+	Points Points
+}
+
+type Points struct {
+	Coordinates [][]float64
+}
+
+func New(baseUrl string, apiKey string) *GraphHopperPathSuplier {
+	return &GraphHopperPathSuplier{apiKey: apiKey, baseUrl: baseUrl}
+}
+
+func (g GraphHopperPathSuplier) FindPath(from m.Location, to m.Location, vehicle string) [][]m.Location {
+	url := g.GetUrlFromLocations(from, to, vehicle)
+	req, _ := http.NewRequest(http.MethodGet, url, nil)
+	c := http.Client{}
+	res, err := c.Do(req)
+	if err != nil || res.StatusCode != 200 {
+		panic(fmt.Sprintf("Request failed: %d, %s", res.StatusCode, res.Body))
+	}
+	return g.DecodeResponseToPath(res)
+}
+
+func (g GraphHopperPathSuplier) DecodeResponseToPath(response *http.Response) [][]m.Location {
```

Comment:
Why this method is public? It will not be used, because it is not present in interface.

Reply:
Did a little refactoring, now it's private